index.js

```diff
@@ -469,6 +476,28 @@ Router.prototype.use = function use(handler) {
     throw new TypeError('argument handler is required')
   }
 
+  var name
+
+  // If a name is used, the last argument will be a string, not a function
+  if (callbacks.length > 1 && typeof callbacks[callbacks.length - 1] !== 'function') {
+    name = callbacks.pop()
+  }
+
+  if(name && !((path instanceof String) || typeof path === 'string')) {
+    throw new Error('only paths that are strings can be named')
+  }
+
+  if(name && this.routes[name]) {
```

Comment:
It looks like we are accepting an empty string as a valid `name` argument, but then silently not acting on it. SHould either just use that as a valid name, or reject it with an error.